packages/firestore/src/local/indexeddb_schema_converter.ts

```diff
@@ -455,6 +468,79 @@ export class SchemaConverter implements SimpleDbSchemaConverter {
       })
       .next(() => PersistencePromise.waitFor(writes));
   }
+
+  // ehsan
+  private runOverlayMigration(
+    db: IDBDatabase,
+    transaction: SimpleDbTransaction
+  ): PersistencePromise<void> {
+    const queuesStore = transaction.store<DbMutationQueueKey, DbMutationQueue>(
+      DbMutationQueueStore
+    );
+    const mutationsStore = transaction.store<
+      DbMutationBatchKey,
+      DbMutationBatch
+    >(DbMutationBatchStore);
+
+    const promises: Array<PersistencePromise<void>> = [];
+    let userIds = new Set<string>();
+
+    return queuesStore
+      .loadAll()
+      .next(queues => {
+        for (const queue of queues) {
+          userIds = userIds.add(queue.userId);
+        }
+      })
+      .next(() => {
+        userIds.forEach(userId => {
```

Comment:
We don't really need the intermediate state (the userIds set). We can directly run the migration in the loop at 491.